src/styles/images/index.md.njk

```diff
@@ -8,53 +8,74 @@ layout: layout-pane.njk
+Alternative text, or alt text, is read out by screen readers or displayed onscreen — usually when images do not load or have been disabled in browser settings. All images should include the alt attribute, even if it’s left empty without alt text (`alt=""`).
 
-{{ example({group: "styles", item: "images", example: "representative", html: false, open: false, size: "l"}) }}
+#### When to use alt text
 
+Only add alt text when you need to describe visual details you cannot practically include in the written content.
 
-## Alternative text
+There are some cases when an empty `alt=""` attribute is more appropriate. These include:
 
-This guidance is for government teams that build online services. [To learn about publishing alternative text in GOV.UK mainstream content, go to GOV.UK's 'Images' guidance.](https://www.gov.uk/guidance/content-design/images)
+- a decorative image that does not include important content
+- an icon with a text label, where the alt text would repeat the text label
+- an image used in a link, where the image is not needed to understand the link
 
-Alternative text, or alt text, is read out by screen readers or displayed if an image does not load or if images have been switched off.
+To decide if an image needs alt text, see the [Web Accessibility Initiative alt decision tree](https://www.w3.org/WAI/tutorials/images/decision-tree/).
 
-All images, except decorative images, must have alt text that:
+#### Writing good alt text
+
+Good alt text:
 
 - tells people what information the image provides
 - describes the content and function of the image
 - is specific, meaningful and concise
 
 Use normal punctuation, like commas and full stops, so the text is easy to read and understand.
 
-Do not:
+Generally, aim for no more than 2 sentences of alt text.
+
+Do not write alt text that:
 
-- include the name of the photographer or person who created the image
-- start with ‘Image of’, ‘Graphic of’ or ‘Photo of’
-- repeat information from the page
-- include extra information not in the image
+- includes the name of the photographer or person who created the image
+- starts with ‘Image of’, ‘Graphic of’ or common information that would repeat across every image
+- repeats information from the page
+- includes extra information not in the image
 
 {{ example({group: "styles", item: "images", example: "alt-text", html: true, open: true, size: "l"}) }}
 
-Some images do not need descriptions. Use `alt=""` if the image is:
+### Avoid images that contain text
+
+Use written content instead of an image that contains text, wherever possible.
+
+If you need to use an image that contains text, create it as a vector graphic (SVG) so that it remains readable when the user zooms in. Text can become pixellated or unreadable when zooming into images, especially PNGs and JPGs.
+
+Any important text you display in an image should be included in the written content — or in alt text if this not possible.
```

Comment:
For this section, we had a suggestion from Anika that should help plug any gaps in the guidance of when text can be in images. @calvin-lau-sig7 care to take a look and see if this addresses some of the pieces we've chatted about?
CC @selfthinker for fact-checking this change, if you're willing.

```suggestion
### Avoid images that contain text

Use written content instead of an image that contains text, wherever possible.

Text that is part of an image can be difficult to read, copy, highlight, resize, re-colour or read out with a screen reader.

Any important text you display in an image should also have written content near the image which conveys the same meaning.

Placing HTML text over top of an image is one way to avoid using an image that contains text, but there will need to be adequate contrast between the text and the image. Ensure there is a contrast ratio of at least 4.5:1 between the text colour and all portions of the image that overlap the text, on all screen sizes.

If you need to use an image that contains text, create it as a vector graphic (SVG) so that it remains readable when the user zooms in. Text can become pixellated or unreadable when zooming into other image file formats, especially PNG and JPG.

If an image that contains text cannot be removed, replaced, or supplemented by written content, then adding alt text that includes the image's text is a helpful last-resort option.

```

Reply:
Considerations for image behind HTML text include:
- what is the contrast with the background if images are unavailable/turned-off
- what happens to the contrast when high-contrast and other colour modes are enabled

Reply:
One more consideration for placing HTML text over an image: When a user resizes the text, make sure the text over the image is also resized, and when it is, make sure it is not overlapping other text or becomes otherwise unreadable (for example, by being white text on a white background).

> Text can become pixellated or unreadable when zooming into other image file formats

A bit nitpicky, but it's not just text that becomes pixellated. The whole image becomes pixellated, which obviously includes the text.

> If an image that contains text cannot be removed, replaced, or supplemented by written content, then adding alt text that includes the image's text is a helpful last-resort option.

If the text is the main purpose of the image, and therefore it counts as an "image of text", and it is not "essential" (as WCAG calls it), then only adding `alt` text is a WCAG fail of 1.4.5 Images of Text.

Reply:
I've taken into account the comments above from @selfthinker and @jbuller and modified the content to add in those points. I did a deep-dive into the exceptions for WCAG criterion 1.4.5 to make sure the guidance we provide aligns with WCAG.

I also reformatted the entire section to add those headings I was mentioning. This greatly expands the content (wow!), but I think it's finally starting to cover all the types of 'images that contain text' that come up in these kinds of conversations.

```suggestion
### Avoid images that contain text

Use written content instead of an image that contains text.

Any important text displayed in an image should also have written content near the image which conveys the same meaning.

####Text that is part of an image

Avoid including text as part of an image, especially for illustrations and charts.

Text can sometimes be an inseparable part of an image, especially in photography. For example, a photo that includes a signpost and a company's name on the side of a building.

Text that is part of an image can be difficult to read, copy, highlight, resize, re-colour or read out with a screen reader.

#### Images of text

Avoid using images of text.

Replace images of text with alternatives whenever possible. For example, a screenshot of a quote or a passage from a book.

When images of text are used, they should have written content near the image which conveys the same meaning. Often the best option is to fully replace the image with HTML text.

Some specific images of text can be used without written content near the image, but must include alt text. These specific images must have a particular presentation of text that is essential to the information being conveyed ([WCAG criterion 1.4.5](https://www.w3.org/WAI/WCAG22/Understanding/images-of-text.html)). For example, a logo containing text or a representation of a letter.

If you need to create an image that contains text, create it as a vector graphic (SVG) whenever possible. Vector graphics remain crisp and readable when the user zooms in. Other image formats like PNG, JPG and JPEG can become pixellated or unreadable when zooming in.

#### Text on top of an image

Placing HTML text onto an image is one way to avoid using images that contain text, but there will need to be adequate contrast between the text and the image.

When placing HTML text onto an image, ensure that:
- there is a contrast ratio of at least 4.5:1 between the text colour and all portions of the image that overlap the text, on all screen sizes
- there is a contrast ratio of at least 4.5:1 between the text colour and the background colour where it sits, in case the image fails to load or is turned off by the user
- the text colour, image and background colour all behave well when high-contrast and other colour modes are enabled
- the text colour, image and background colour work well for various forms of colourblindness
```

Reply:
Given the difficulties of getting it right, and easy of getting it wrong perhaps 'Text on top of an image' should be marked as 'Avoid' too?

> Avoid including text as part of an image, especially for illustrations and charts.
So recommending against graphs, charts, flowcharts, diagrams even if there's equivalent body text? That's not great for dyslexia etc.

I feel the paragraph about logos could be reordered to be more concise.

Reply:
Hey @jbuller! I've got some responses and clarifying questions for those 3 latest notes you added here.

1. I'm thinking instead of saying to avoid text on top of an image, we could say to 'be cautious when' or something? Text on top of an image can be worlds better than many of the other options here, but it does need to be done carefully.

2. When saying "Avoid including text as part of an image, especially for illustrations and charts" I was mainly thinking that portions of the chart that can be pulled out of the image and made into text should be pulled out. For example, a chart title can be in HTML text, as can any stuff that would previously be 'fine print' in an image of a chart. Would adding something more specific about what was meant by "especially for illustrations and charts" make sense?

3. Do you have a suggestion for reorganising the paragraph that includes a logo as an example? That paragraph is 3 sentences at the moment: a statement, followed by clarification of what 'specific images' means, and then concluded with examples. For all the other paragraphs in this section, I've tried to keep the "for example" part at the end of the paragraph.

Reply:
2. Yes you could add a clarification about titles and footnotes. I just think particularly that for those who can't make vector graphics, it should be ok to have a (flow)chart along with equivalent text. As it stands the axis labels of a graph etc ban its use.
The original
> Use written content instead of an image that contains text, wherever possible.
was clearer that the subsections set out the exceptions 

3. Consider
> If the exact appearance of the text in the image is essential, it must include alt text, but equivalent written content nearby is not required: ([WCAG criterion 1.4.5](https://www.w3.org/WAI/WCAG22/Understanding/images-of-text.html)). For example, letters in a logo or a handwritten letter.

(Note that I interpreted your words 'representation of a letter' as meaning a stylised character rather than paper correspondence until I read the WCAG page with its explanation)

Reply:
Alright, I think these are some good notes on the remaining ambiguity, thanks!

I've added the "be cautious when..." part that I mentioned in point 1.

```
#### Text on top of an image
Placing HTML text onto an image is one way to avoid using images that contain text, but there will need to be adequate contrast between the text and the image.
Be cautious when placing HTML text on top of an image, and be sure to test that:...
```

Based on conversations happening around our illustration guidance, I think we're going to steer clear of giving too much detail on images of charts. The overall category that includes charts also includes other related things like diagrams and graphs. All of these aren't inherently 'images', since they can also be made in HTML or JavaScript. In the other update we're working on, we may just note that charts are part of a larger pattern that goes beyond the scope of image guidance.

Here's the new section that mentions the word 'chart'

```
Avoid including text as part of an image, especially for illustrations.
For charts, diagrams and graphics, use HTML for as many elements as is feasible. This includes the chart title and any footnotes.
```

I'll also change 'letter' to 'memo' so that all ambiguity is removed. A handwritten letter could still technically mean a single letter of the alphabet.

There are specific types of images where the exact appearance of the text in the image is essential to its purpose. These types of images must include alt text, but equivalent written content nearby is not required ([WCAG criterion 1.4.5](https://www.w3.org/WAI/WCAG22/Understanding/images-of-text.html)). For example, a logo or a handwritten memo.

Here's an updated version, which tries to clean up the ambiguity:

```suggestion
### Avoid images that contain text
Use written content instead of an image that contains text.
Any important text displayed in an image should also have written content near the image which conveys the same meaning.
####Text that is part of an image
Avoid including text as part of an image, especially for illustrations.
For charts, diagrams and graphics, use HTML for as many elements as is feasible. This includes the chart title and any footnotes.
Text can sometimes be an inseparable part of an image, especially in photography. For example, a photo that includes a signpost and a company's name on the side of a building.
Text that is part of an image can be difficult to read, copy, highlight, resize, re-colour or read out with a screen reader.
#### Images of text
Avoid using images of text.
Replace images of text with alternatives whenever possible. For example, a screenshot of a quote or a passage from a book.
When images of text are used, they should have written content near the image which conveys the same meaning. Often the best option is to fully replace the image with HTML text.
##### Exceptions for images of text
There are specific types of images where the exact appearance of the text in the image is essential to its purpose. These types of images must include alt text, but equivalent written content nearby is not required ([WCAG criterion 1.4.5](https://www.w3.org/WAI/WCAG22/Understanding/images-of-text.html)). For example, a logo or a handwritten memo.
If you need to create an image that contains text, create it as a vector graphic (SVG) whenever possible. Vector graphics remain crisp and readable when the user zooms in. Other image formats like PNG, JPG and JPEG can become pixellated or unreadable when zooming in.
#### Text on top of an image
Placing HTML text onto an image is one way to avoid using images that contain text, but there will need to be adequate contrast between the text and the image.
Be cautious when placing HTML text on top of an image, and be sure to test that:
- there is a contrast ratio of at least 4.5:1 between the text colour and all portions of the image that overlap the text, on all screen sizes
- there is a contrast ratio of at least 4.5:1 between the text colour and the background colour where it sits, in case the image fails to load or is turned off by the user
- the text colour, image and background colour all behave well when high-contrast and other colour modes are enabled
- the text colour, image and background colour work well for various forms of colourblindness
```

Reply:
Hey @calvin-lau-sig7 how does this latest draft of the "images and text" section of the alt text guidance look? Do you think it's ready to commit?

Reply:
One more edit, to simplify the info in the 'Exceptions for images of text' section.

I also added a sub-heading for using vector graphics for images of text. It was getting unintentionally buried under the 'Exceptions' heading.

I've also bumped the heading levels up one, to give this section a 2nd level heading.

```suggestion
## Avoid images that contain text

Use written content instead of an image that contains text.
Any important text displayed in an image should also have written content near the image which conveys the same meaning.

###Text that is part of an image

Avoid including text as part of an image, especially for illustrations.
For charts, diagrams and graphics, use HTML for as many elements as is feasible. This includes the chart title and any footnotes.

Text can sometimes be an inseparable part of an image, especially in photography. For example, a photo that includes a signpost and a company's name on the side of a building.
Text that is part of an image can be difficult to read, copy, highlight, resize, re-colour or read out with a screen reader.

### Images of text

Avoid using images of text.
Replace images of text with alternatives whenever possible. For example, a screenshot of a quote or a passage from a book.
When images of text are used, they should have written content near the image which conveys the same meaning. Often the best option is to fully replace the image with HTML text.

#### Exceptions: logos and handwritten memos

There are specific types of images, such as logos and handwritten notes, where the exact appearance of the text in the image is essential to its purpose. These types of images must include alt text, but do not require written content near the image which conveys the same meaning ([WCAG criterion 1.4.5](https://www.w3.org/WAI/WCAG22/Understanding/images-of-text.html)).

#### Using vector graphics for images of text
If you need to create an image that contains text, create it as a vector graphic (SVG) whenever possible. Vector graphics remain crisp and readable when the user zooms in. Other image formats like PNG, JPG and JPEG can become pixellated or unreadable when zooming in.

### Text on top of an image

Placing HTML text onto an image is one way to avoid using images that contain text, but there will need to be adequate contrast between the text and the image.
Be cautious when placing HTML text on top of an image, and be sure to test that:
- there is a contrast ratio of at least 4.5:1 between the text colour and all portions of the image that overlap the text, on all screen sizes
- there is a contrast ratio of at least 4.5:1 between the text colour and the background colour where it sits, in case the image fails to load or is turned off by the user
- the text colour, image and background colour all behave well when high-contrast and other colour modes are enabled
- the text colour, image and background colour work well for various forms of colourblindness
```